### ISSUE TITLE
Unstable UT TestGenerateTableSinkTaskWithBarrierTs in TiFlow 

### ISSUE THREAD

**Issue by hongyunyan** [pingcap/tiflow#10523]
### What did you do?

run ut in ci
![image](https://github.com/pingcap/tiflow/assets/26538495/b988e23f-a856-4272-811b-0e6afee0f898)

### What did you expect to see?

run test success

### What did you see instead?

Failed many times and succeeded sometimes.

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
```

TiCDC version (execute `cdc version`):

```console
both master and 6.5
```

**Comment by purelind** [comment]
also found this unstable ut on release-7.1

**Comment by asddongmen** [comment]
closed by #10770
